### DISCUSSION TITLE
How to install zwave-js-ui in debian 32bits?

### DISCUSSION THREAD

**Discussion by jotakar** [zwave-js/zwave-js-ui#2854] [Q&A]
Hi, when I try to install ZWAVE_JS-UI using span I get a message telling that there is not a  package available for my architecture:  Debian 11 in a 32bits machine.
So, I wonder if there would be any other way to install ZWave-JS-UI in a 32bits machine.
Thanks

**Comment by robertsLando** [comment]
@jotakar Did you tried with docker? What's the output of the command `uname -a` ?

**Reply by jotakar** [reply]
The problem is that my system is a 32bits architecture, so no docker. And I don't know enough about Docker. I'ww keep on looking for a solution. Thanks.

**Reply by robertsLando** [reply]
Snap should support armhf too, tried it? @jmgiaever am I right?

**Reply by robertsLando** [reply]
@jotakar cold you tell me the output of `uname -a` ?

**Reply by jmgiaever** [reply]
The snap has armhf yes

**Reply by jotakar** [reply]
No, it's not raspberry but a system based on Intel with Debian 11 installed
**unmae** results in 
```
Linux domo 5.10.0-19-686-pae #1 SMP Debian 5.10.149-2 (2022-10-21) i686 GNU/Linux
```
As you can see iits a 32 bits processor. 
And lsb_release result is
```
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye
A Debian 11 avalaible for this platform. 
```
I'm afraid it's an very ancient machine, so It cannot make great efforts  :)

**Reply by robertsLando** [reply]
The problem is it is `i686` that is a very old cpu. AFAIK nodejs itsef stopped to support it from nodejs 10+ and zui requires at least node 14 to run. 

You should check if you find some unofficail node dist that supporots i686 and manually build it locally:

```
git clone https://github.com/zwave-js/zwave-js-ui
cd zwave-js-ui
yarn install
yarn run build
yarn start
```
